src/prefect/core/parameter.py

```diff
@@ -112,3 +114,29 @@ def serialize(self) -> Dict[str, Any]:
             - dict representing this parameter
         """
         return prefect.serialization.task.ParameterSchema().dump(self)
+
+
+class DateTimeParameter(Parameter):
+    """
+    A DateTimeParameter that casts its input as a DateTime
+    Args:
+        - name (str): the Parameter name.
+        - required (bool, optional): If True, the Parameter is required and the default
+            value is ignored.
+        - default (any, optional): A default value for the parameter. If the default
```

Comment:
Hmmm can you actually provide a `default` to this class?  This could be confusing - I think you can provide a default but it needs to be an already serialized datetime object.  We should either document this or just not let you provide a default to this parameter type?

Reply:
I'm sure this is too wacky for y'all but in my custom version of this I had it default to `datetime.datetime.now()` 😄

Reply:
At one time we did allow or plan to allow `callable` defaults, or am I off?

Reply:
> I think you can provide a default but it needs to be an already serialized datetime object.

Why is this true? Couldn't the serializer be modified to take care of this automatically? I'd find forcing the default to be different than the wrapped type non-intuitive.

Reply:
With the implementation in this PR, this attribute will be serialized to Cloud using this logic: https://github.com/PrefectHQ/prefect/blob/master/src/prefect/serialization/task.py#L123 which requires the default to be JSON compatible. 

We could introduce a whole new serializer for this new class, but that seems like a slippery slope to me.

And yea, I agree that the behavior of a `default` on this class would be wonky as-currently-implemented - I propose removing the ability to specify a default on this class.

Reply:
I removed the ability to specify any custom default (but allowed `None` so the parameter can be optional)